### PR TITLE
Make frontend nice

### PR DESCRIPTION
**Pull request by danielperev**
yeah, its nice now.

### CODE REVIEW

File: frontend/src/reusable/inputs/PromptInputMapping.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import TextPrompt from "./TextPrompt";
+
+export const PromptMapping: Map<
+    string,
+    (props: PromptDefinition & SendUpdateToCollector) => React.ReactNode
+> = new Map([
+    ["text", TextPrompt],
+    ["listbox", ListBoxPrompt],
+
+    // "button" : ,
```
Comment (danielperev) [inline frontend/src/reusable/inputs/PromptInputMapping.ts:12]: will add some new ones (datetime/radio) in next PR

File: frontend/src/reusable/default/DefaultPromptDefinitions.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        displayText: "What is your job title?",
+        idKey: "job_title",
+        inputType: "radio",
+    },
+    {
+        displayText:
+            "How many years of professional experience in your field did you have prior to accepting your job offer?",
+        idKey: "num_years_exp",
+        inputType: "radio",
```
Comment (pecasdev) [inline frontend/src/reusable/default/DefaultPromptDefinitions.ts:46]: these shouldn't be radio, they should be sliders

File: frontend/src/reusable/inputs/ListBoxPrompt.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { ChevronDownIcon, CheckIcon } from "@heroicons/react/20/solid";
+
+export default function ListBoxPrompt(props: SimplePromptDefinition) {
+    let currentSelected = props.state.selected;
+    return (
+        <Listbox
+            onChange={(e) => {
+                props.stateManager(e);
+                if (props.doSomething) props.doSomething(e);
+            }}
```
Comment (pecasdev) [inline frontend/src/reusable/inputs/ListBoxPrompt.tsx]: yucky!

Comment (pecasdev) [reply frontend/src/reusable/inputs/ListBoxPrompt.tsx]: if you make doSomething non optional, you don't have to do this anymore

**Review by pecasdev — APPROVED** [review]

File: frontend/src/reusable/default/DefaultPromptDefinitions.ts (hunk 2 above)
Comment (danielperev) [reply frontend/src/reusable/default/DefaultPromptDefinitions.ts:46]: neither are in, i will fix once both are in, noted.

File: frontend/src/reusable/inputs/ListBoxPrompt.tsx (hunk 3 above)
Comment (danielperev) [reply frontend/src/reusable/inputs/ListBoxPrompt.tsx]: honestly i think passing down this function is wrong in the first place and theres probably a better way of doing it but youre right ill make it required

**Pull request merged**